Assets/HoloToolkit/SpatialSound/Scripts/AudioLoFiEffect.cs

```diff
@@ -0,0 +1,222 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+    /// <summary>
+    /// An audio effect that limits the frequency range of a sound to simulate
+    /// being played over various telephony or radio sources.
+    /// </summary>
+    /// <remarks>
+    /// For the best results, also attach an AudioInfluencerManager to the sound
+    /// source. This will ensure that the proper frequencies will be restored
+    /// when audio influencers are used in the scene.
+    /// </remarks>
+    [RequireComponent(typeof(AudioSource))]
+    [RequireComponent(typeof(AudioLowPassFilter))]
+    [RequireComponent(typeof(AudioHighPassFilter))]
+    [DisallowMultipleComponent]
+    public class AudioLoFiEffect : MonoBehaviour
+    {
+        /// <summary>
+        /// The quality level of the simulated audio source (ex: AM radio).
+        /// </summary>
+        [Tooltip("The quality level of the simulated audio source.")]
+        public AudioSourceQuality SourceQuality;
+
+        // The audio filter settings that match the selected source quality.
+        private FilterSettings filterSettings;
+
+        // The filters used to simulate the source quality.
+        private AudioLowPassFilter lowPassFilter;
+        private AudioHighPassFilter highPassFilter;
+
+        // Collection used to look up the filter settings that match the selected
+        // source quality.
+        private Dictionary<AudioSourceQuality, FilterSettings> sourceQualityFilterSettings =
+            new Dictionary<AudioSourceQuality, FilterSettings>();
+
+        private void Awake()
+        {
+            LoadQualityFilterSettings();
+
+            filterSettings = sourceQualityFilterSettings[SourceQuality];
+
+            lowPassFilter = gameObject.GetComponent<AudioLowPassFilter>();
+            lowPassFilter.cutoffFrequency = filterSettings.LowPassCutoff;
+
+            highPassFilter = gameObject.GetComponent<AudioHighPassFilter>();
+            highPassFilter.cutoffFrequency = filterSettings.HighPassCutoff;
+        }
+
+        private void Update()
+        {
+            FilterSettings fs = sourceQualityFilterSettings[SourceQuality];
+            if (fs != filterSettings)
+            {
+                // If we have an attached AudioInfluencerManager, we need to let it know
+                // about our filter settings change.
+                AudioInfluencerManager influencerManager = gameObject.GetComponent<AudioInfluencerManager>();
```

Comment:
If this is a Manager, then shouldn't we be able to get it's singleton instance instead of getting the component?

Reply:
This is a component level manager. It manages the influencers that are effecting the AudioSource attached to the game object. I can rename it (in vNext) to be a Controller if that seems better.